spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -47,13 +55,20 @@ public class AnalyzeTableSparkAction extends BaseSparkAction<AnalyzeTableSparkAc
   private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
 
   private final Table table;
-  private Set<String> columns = ImmutableSet.of();
-  private Set<String> types = StandardBlobTypes.blobTypes();
+  private final Set<String> supportedBlobTypes =
+      ImmutableSet.of(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1);
+  private Set<String> columns;
+  private Set<String> blobTypesToAnalyze = supportedBlobTypes;
   private Long snapshotId;
 
   AnalyzeTableSparkAction(SparkSession spark, Table table) {
     super(spark);
     this.table = table;
+    Snapshot snapshot = table.currentSnapshot();
+    ValidationException.check(snapshot != null, "Cannot analyze a table that has no snapshots");
```

Comment:
It would be nice to handle this case gracefully.
Table without snapshots is an empty table (no data).
Also, stats are assigned to snapshots, and there is no snapshot, so there cannot be a stats file created. 
Thus there is only one way to handle this gracefully -- just no-op.
I believe this would be better from user-perspective than just throwing.